cpp/src/io/parquet/page_data.cu

```diff
@@ -51,6 +51,11 @@ constexpr int non_zero_buffer_size = block_size * 2;
 
 constexpr int rolling_index(int index) { return index & (non_zero_buffer_size - 1); }
 
+// Use up to 512 bytes of shared memory as a cache for nesting information.
+// As of 1/20/23, this gives us a max nesting depth of 10 (after which it falls back to
+// global memory). This handles all but the most extreme cases.
+constexpr int max_cacheable_nesting_decode_info = (512) / sizeof(PageNestingDecodeInfo);
```

Comment:
Do we have any tests in C++ or Python that cover the fallback (nesting depth > 10)?

Reply:
Done.